tests/test_preprocess.py

```diff
@@ -75,3 +81,92 @@ def test_build_vocab(self):
         assert set(vocab.get_index_to_token_vocabulary("chars").values()) == set(
             ["@@PADDING@@", "@@UNKNOWN@@", "a", "b", "c"]
         )
+
+
+class TestModelPreprocessingInterface(unittest.TestCase):
+    def test_max_tokens_limited_by_small_max_seq_len(self):
+        args = config.Params(max_seq_len=7, input_module="bert-base-uncased")
+        mpi = ModelPreprocessingInterface(args)
+        self.assertEqual(mpi.max_tokens, 7)
+
+    def test_max_tokens_limited_by_bert_model_max(self):
+        args = config.Params(max_seq_len=None, input_module="bert-base-uncased")
+        mpi = ModelPreprocessingInterface(args)
+        self.assertEqual(mpi.max_tokens, 512)
+
+    def test_boundary_token_fn_trunc_w_default_strategies(self):
+        MAX_SEQ_LEN = 7
+        args = config.Params(max_seq_len=MAX_SEQ_LEN, input_module="bert-base-uncased")
+        mpi = ModelPreprocessingInterface(args)
+        seq = mpi.boundary_token_fn(
+            ["Apple", "buy", "call"],
+            s2=["Xray", "you", "zoo"],
+            trunc_strategy="trunc_s2",
+            trunc_side="right",
+        )
+        self.assertEqual(len(seq), MAX_SEQ_LEN)
+        self.assertEqual(seq, ["[CLS]", "Apple", "buy", "call", "[SEP]", "Xray", "[SEP]"])
+
+    def test_boundary_token_fn_trunc_s1_left_side(self):
+        MAX_SEQ_LEN = 7
+        args = config.Params(max_seq_len=MAX_SEQ_LEN, input_module="bert-base-uncased")
+        mpi = ModelPreprocessingInterface(args)
+        seq = mpi.boundary_token_fn(
+            ["Apple", "buy", "call"],
+            s2=["Xray", "you", "zoo"],
+            trunc_strategy="trunc_s1",
+            trunc_side="left",
+        )
+        self.assertEqual(len(seq), MAX_SEQ_LEN)
+        self.assertEqual(seq, ["[CLS]", "call", "[SEP]", "Xray", "you", "zoo", "[SEP]"])
+
+    def test_boundary_token_fn_trunc_s1_left_side(self):
```

Comment:
duplicate test?

Reply:
I'm guessing this should left -> right?

Reply:
Thanks — yeah, should have been `right`. Fixing with next commit.